app/src/androidTest/java/com/github/se/project/ui/endToEnd/TutorEndToEnd.kt

```diff
@@ -0,0 +1,251 @@
+    doReturn(mockLessonFlow).`when`(mockLessonViewModel).selectedLesson
+    mockLessonFlow.value = mockLessons[0]
+  }
+
+  @Test
+  fun TutorEndToEndTest() {
+    composeTestRule.setContent {
+      PocketTutorApp(true, viewModel(), mockProfileViewModel, mockLessonViewModel)
+    }
+    // Thread.sleep(50000)
+    // Sign In Screen
+    composeTestRule.onNodeWithTag("logo").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("loginButton").performClick()
+    composeTestRule.onNodeWithTag("firstNameField").assertIsDisplayed()
+
+    // Create Profile Screen
+    composeTestRule.onNodeWithTag("firstNameField").performTextInput("John")
+    composeTestRule.onNodeWithTag("lastNameField").performTextInput("Doe")
+    composeTestRule.onNodeWithTag("phoneNumberField").performTextInput("0213456789")
+    composeTestRule.onNodeWithTag("roleButtonTutor").performClick()
+    composeTestRule.onNodeWithTag("sectionDropdown").performClick()
+    composeTestRule.onNodeWithTag("sectionDropdownItem-SC").performClick()
+    composeTestRule.onNodeWithTag("academicLevelDropdown").performClick()
+    composeTestRule.onNodeWithTag("academicLevelDropdownItem-BA3").performClick()
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+    assertEquals(Role.TUTOR, mockProfileViewModel.currentProfile.value?.role)
+
+    // Create Tutor Profile Screen
+    composeTestRule.onNodeWithTag("welcomeText").assertIsDisplayed()
+    composeTestRule.onNodeWithTag("checkbox_FRENCH").performClick()
+    composeTestRule.onNodeWithTag("checkbox_ENGLISH").performClick()
+    composeTestRule.onNodeWithTag("subjectButton").performClick()
+    composeTestRule.onNodeWithTag("dropdownANALYSIS").performClick()
+    composeTestRule.onNodeWithTag("dropdownPHYSICS").performClick()
+    composeTestRule.onNodeWithTag("subjectButton").performClick()
+    composeTestRule.onNodeWithTag("priceSlider").performGesture { swipeRight() }
+    composeTestRule.onNodeWithTag("confirmButton").performClick()
+
+    // Create Tutor Schedule Screen
+    composeTestRule
+        .onNodeWithTag("welcomeText")
+        .assertTextEquals("John, show us your availabilities")
+    composeTestRule.onNodeWithTag("Slot_0_0").performClick()
+    composeTestRule.onNodeWithTag("Slot_0_3").performClick()
+    composeTestRule.onNodeWithTag("Slot_0_2").performClick()
+    composeTestRule.onNodeWithTag("Slot_0_6").performClick()
+    composeTestRule.onNodeWithTag("FindStudentButton").performClick()
+
+    // Home Screen
+    composeTestRule.onNodeWithContentDescription("Profile Icon").performClick()
+    composeTestRule.onNodeWithTag("profileStatus").assertTextEquals("Status: BA3 Tutor")
+    composeTestRule.onNodeWithTag("profileSection").assertTextEquals("Section: SC")
+    composeTestRule.onNodeWithTag("profilePrice").assertTextEquals("Price: 50.- per hour")
+    composeTestRule
+        .onNodeWithTag("lessonsCount")
+        .assertTextEquals("0 lessons given since you joined PocketTutor")
```

Comment:
In the interest of testing all the features, it would be nice if the test also edited the profile and/or schedule